File: backend/src/main/java/net/dnadas/training_portal/dto/group/project/questionnaire/QuestionnaireSubmissionResponseDetailsDto.java
package net.dnadas.training_portal.dto.group.project.questionnaire;

import java.util.List;

public record QuestionnaireSubmissionResponseDetailsDto(
  Long id, String name, String description,
  List<SubmittedQuestionResponseDto> questions,
  Integer receivedPoints, Integer maxPoints, String createdAt) {
}

File: frontend/src/questionnaires/dto/QuestionnaireSubmissionResponseDetailsDto.ts
import {SubmittedQuestionResponseDto} from "./SubmittedQuestionResponseDto.ts";

export interface QuestionnaireSubmissionResponseDetailsDto {
  id: number;
  name: string;
  description: string;
  receivedPoints: number;
  maxPoints: number;
  createdAt: string;
  questions: SubmittedQuestionResponseDto[];
}
